pegasus/forms/petition.rb

```diff
@@ -45,4 +40,34 @@ def self.process(data)
   def self.receipt
     'petition_receipt'
   end
+
+  # @params zip_code_or_country [String] a five digit US zip code or country
+  #   name.
+  # @returns [Hash] data about zip_code_or_country.
+  def self.get_location(zip_code_or_country)
+    # Note that, empirically, we see non-US zip codes so we check for
+    # numerality and length.
+    is_us_zip_code = zip_code_or_country.length == 5 &&
+      zip_code_or_country == zip_code_or_country.to_i.to_s
+    return {country_s: zip_code_or_country} unless is_us_zip_code
```

Comment:
This could be simplified with a regex: `is_us_zip_code = !!(/^\d{5}$/ =~ zip_code_or_country)`

Or even
```ruby
return {country_s: zip_code_or_country} unless /^\d{5}$/ =~ zip_code_or_country
```